### PR TITLE
(Strongly Typed) Amplitude client.

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

Adding the Amplitude NodeJS (so backend only) SDK.
A few things to know:
- It's a strongly typed client based on our events and properties definitions from the Amplitude UI.
- You basically define some events on the Amplitude UI, and then download your types via `front>npx ampli pull -p ./lib/utils/ampli --omit-api-keys`
- I created a super small layer that returns a typed dev or prod client depending on the env vars, to avoid talking to Amplitude in dev and forcing engineers of the team to defined useless env vars.
- I am planning on using this for my ongoing task, which is the onboarding "[quick guide](https://github.com/dust-tt/dust/issues/4014)", hence this PR now.

## Deploy Plan

- Just deploy front
- Add the AMPLITUDE_ENV and AMPLITUDE_API_KEY when we want to start tracking.

### CODE REVIEW

File: front/ampli.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (lasryaric) [inline front/ampli.json:1]: Unfortunately this has to be in the repo, but there is no sensitive information in there.

Comment (flvndvd) [reply front/ampli.json:1]: Based on this, it seems doable: https://github.com/aws-amplify/amplify-swift/issues/1662#issuecomment-1059564469

File: front/lib/utils/ampli/index.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  /**
+   * Initialize the Ampli SDK. Call once when your application starts.
+   *
+   * @param options Configuration options to initialize the Ampli SDK with.
+   */
+  load(options: LoadOptions): PromiseResult<void> {
+    this.disabled = options.disabled ?? false;
+
+    if (this.amplitude) {
+      console.warn('WARNING: Ampli is already initialized. Ampli.load() should be called once at application startup.');
```
Comment (flvndvd) [inline front/lib/utils/ampli/index.ts:125]:
```suggestion
      logger.warn('WARNING: Ampli is already initialized. Ampli.load() should be called once at application startup.');
```

File: front/lib/utils/ampli/index.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    return this.amplitude!;
+  }
+
+  get isLoaded(): boolean {
+    return this.amplitude != null;
+  }
+
+  private isInitializedAndEnabled(): boolean {
+    if (!this.amplitude) {
+      console.error('ERROR: Ampli is not yet initialized. Have you called ampli.load() on app start?');
```
Comment (flvndvd) [inline front/lib/utils/ampli/index.ts:110]: Let's use `logger`.

File: front/lib/utils/ampli/index.ts [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    }
+
+    if (options.client && 'instance' in options.client) {
+      this.amplitude = options.client.instance;
+    } else if (apiKey) {
+      this.amplitude = amplitude.createInstance();
+      const configuration = (options.client && 'configuration' in options.client) ? options.client.configuration : {};
+      return this.amplitude.init(apiKey, { ...DefaultConfiguration, ...configuration });
+    } else {
+      console.error("ERROR: ampli.load() requires 'environment', 'client.apiKey', or 'client.instance'");
```
Comment (flvndvd) [inline front/lib/utils/ampli/index.ts:143]: logger

File: front/lib/utils/amplitude_client.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+// Get the appropriate Amplitude instance based on the environment variables.
+// The amplitude client is strongly typed based on our events and properties definitions
+// in the Amplitude UI, but we don't want to actually talk to Amplitude in dev.
+export function getAmplitude(): Ampli {
+  if (AmplitInstance) {
+    return AmplitInstance;
+  }
+  const { AMPLITUDE_ENV, AMPLITUDE_API_KEY } = process.env;
+  if (!AMPLITUDE_ENV || !AMPLITUDE_API_KEY) {
```
Comment (flvndvd) [inline front/lib/utils/amplitude_client.ts:32]: Can we stick to one variable to define the current environment? We already have `NODE_ENV`.

File: front/lib/utils/amplitude_client.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    logger.debug("Amplitude environment variables not set, using dev instance");
+    AmplitInstance = new AmpliDev();
+  } else {
+    console.log("Amplitude environment variables set, using prod instance");
```
Comment (flvndvd) [inline front/lib/utils/amplitude_client.ts:36]: Let's use logger here as well.

File: front/ampli.json (hunk 1 above)
Comment (lasryaric) [reply front/ampli.json:1]: @flvndvd the example shown here is about a different project. I think it's completely unrelated to that.
Here is the correct npmjs package: https://www.npmjs.com/package/@amplitude/ampli

This "Ampli.json" object is here to tell the `npx ampli` CLI where we are in terms of properties and event versions types synchronization.

File: front/lib/utils/ampli/index.ts (hunk 3 above)
Comment (lasryaric) [reply front/lib/utils/ampli/index.ts:110]: Can't use logger here, it's going to be overridden each time we pull from Amplitude (ie: each time we define a new event or property).

File: front/lib/utils/ampli/index.ts (hunk 2 above)
Comment (lasryaric) [reply front/lib/utils/ampli/index.ts:125]: Same. Generated code.

File: front/lib/utils/amplitude_client.ts (hunk 5 above)
Comment (lasryaric) [reply front/lib/utils/amplitude_client.ts:32]: This var env is different, it maps to Amplitude enviroment, which would be called "dust-dev" or "dust-prod". For now its just `dust`

**Review by flvndvd — APPROVED** [review]

**Pull request merged**